packages/grafana-data/src/transformations/transformers/histogram.ts

```diff
@@ -152,7 +152,7 @@ export function buildHistogram(frames: DataFrame[], options?: HistogramTransform
   let bucketOffset = options?.bucketOffset ?? 0;
 
   // if bucket size is auto, try to calc from all numeric fields
-  if (!bucketSize) {
+  if (!bucketSize || bucketSize < 0) {
```

Comment:
weirdly.... I think this (and the changes in public/app/plugins/panel/histogram/module.tsx) should be in their own PR.

Otherwise everything else is around making the numer input behave rationally (nothing to do with histogram)